packages/dd-trace/src/ci-visibility/exporters/agent-proxy/index.js

```diff
@@ -3,37 +3,32 @@
 const AgentWriter = require('../../../exporters/agent/writer')
 const AgentlessWriter = require('../agentless/writer')
 const CoverageWriter = require('../agentless/coverage-writer')
-const AgentInfoExporter = require('../../../exporters/common/agent-info-exporter')
-const log = require('../../../log')
+const CiVisibilityExporter = require('../ci-visibility-exporter')
 
 const AGENT_EVP_PROXY_PATH = '/evp_proxy/v2'
 
 function getIsEvpCompatible (err, agentInfo) {
   return !err && agentInfo.endpoints.some(url => url.includes(AGENT_EVP_PROXY_PATH))
 }
 
-const getIsTestSessionTrace = (trace) => {
-  return trace.some(span =>
-    span.type === 'test_session_end' || span.type === 'test_suite_end'
-  )
-}
-
-/**
- * AgentProxyCiVisibilityExporter extends from AgentInfoExporter to get the agent information.
- * If the agent is event platform proxy compatible,
- * it will initialize the AgentlessWriter and CoverageWriter, else it will fall back to AgentWriter.
- */
-class AgentProxyCiVisibilityExporter extends AgentInfoExporter {
+class AgentProxyCiVisibilityExporter extends CiVisibilityExporter {
   constructor (config) {
     super(config)
 
-    this._coverageBuffer = []
-    const { tags, prioritySampler, lookup, protocolVersion, headers } = config
+    const {
+      tags,
+      prioritySampler,
+      lookup,
+      protocolVersion,
+      headers,
+      isGitUploadEnabled
+    } = config
 
     this.getAgentInfo((err, agentInfo) => {
       this._isInitialized = true
-      this._isEvpCompatible = getIsEvpCompatible(err, agentInfo)
-      if (this._isEvpCompatible) {
+      const isEvpCompatible = getIsEvpCompatible(err, agentInfo)
+      this._isUsingEvpProxy = true
```

Comment:
Is this ok here? even when `isEvpCompatible` is false?

Reply:
yeah good catch. It's no issue because `isEvpCompatible` is false we're not calling any of the ITR APIs, but it should still be fixed